### PR TITLE
Make FunctionalTensorWrapper correctly handle symbolic shapes

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85975
* #85858

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: aten/src/ATen/FunctionalTensorWrapper.cpp [hunk 1]
```diff
@@ -206,12 +206,7 @@ void FunctionalTensorWrapper::replace_(const Tensor& other) {
   value_ = other;
   // out= ops are allowed to resize the output tensors, mutating both the data and metadata of the tensor.
   // We need to propagate that metadata mutation to the wrapper (new size).
-  if (sizes() != value_.sizes() || strides() != value_.strides()) {
```
Comment (bdhirsh) [inline aten/src/ATen/FunctionalTensorWrapper.cpp:209]: I remember adding this check a while ago to fix a failing test, and I forget the reason (sigh). But I'm not too concerned if CI is green (and it makes sense to me not to want this for SymInts)

**Review by bdhirsh — APPROVED** [review]

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**